modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/CheckboxTag.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.taglib.util.TagResourceBundleUtil;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Kresimir Coko
+ */
+public class CheckboxTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		if (Validator.isNotNull(_label)) {
+			setDynamicAttribute(StringPool.BLANK, "aria-label", _label);
+		}
+
+		return super.doStartTag();
+	}
+
+	public String getLabel() {
+		return _label;
+	}
+
+	public String getName() {
+		return _name;
+	}
+
+	public String getValue() {
+		return _value;
+	}
+
+	public boolean isChecked() {
+		return _checked;
+	}
+
+	public boolean isDisabled() {
+		return _disabled;
+	}
+
+	public boolean isIndeterminate() {
+		return _indeterminate;
+	}
+
+	public boolean isInline() {
+		return _inline;
+	}
+
+	public void setChecked(boolean checked) {
+		_checked = checked;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setIndeterminate(boolean indeterminate) {
+		_indeterminate = indeterminate;
+	}
+
+	public void setInline(boolean inline) {
+		_inline = inline;
+	}
+
+	public void setLabel(String label) {
+		_label = label;
+	}
+
+	public void setName(String name) {
+		_name = name;
+	}
+
+	public void setValue(String value) {
+		_value = value;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_checked = false;
+		_disabled = false;
+		_indeterminate = false;
+		_inline = false;
+		_label = null;
+		_name = null;
+		_value = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/Checkbox";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("checked", _checked);
+		props.put("disabled", _disabled);
+		props.put("indeterminate", _indeterminate);
+		props.put("inline", _inline);
+
+		if (Validator.isNotNull(_label)) {
+			props.put(
+				"label",
+				LanguageUtil.get(
```

Comment:
Why are we translating this only here? (`_label` it's being used for `aria-label` and in SSR-line 189)

Reply:
I saw that in one of the other examples like this one: [ButtonTag.java](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ButtonTag.java#L273-L279), so used that.

I guess the goal is to **let users handle translations** instead of us doing it in implementations like this one?

Reply:
Yep, I mean why only translate it for the props, we're using `_label` for the `aria-label` and rendering it in `processStartTag`  too

Reply:
So, why not translate it in `getLabel` for example and call `getLabel` in those spots instead `_label`?

Reply:
That would follow the pattern we have with `getId` and `getCssClass`, so I'll do that, sure.